src/chain/sync-manager.js

```diff
@@ -0,0 +1,285 @@
+    // get latest current child block
+    let [childBlockNumber, latestBlockDetails] = await Promise.all([
+      this.chain.parentContract.methods.currentChildBlock().call(),
+      this.chain.getLatestHead()
+    ])
+    childBlockNumber = new BN(childBlockNumber)
+
+    let storedBlockNumber = new BN(0)
+    if (latestBlockDetails) {
+      storedBlockNumber = new BN(utils.toBuffer(latestBlockDetails.number))
+    }
+
+    if (storedBlockNumber.add(new BN(1)).lt(childBlockNumber)) {
+      this._syncBlocks(storedBlockNumber.add(new BN(1)).toNumber())
+    } else {
+      // No new blocks to sync yet
+      this.syncing = false
+    }
+  }
+
+  broadcastNewTx(txBytes, excluded) {
+    const message = JSON.stringify({
+      type: 'TX:ADD',
+      from: this.hostString,
+      data: txBytes
+    })
+
+    // broadcast message
+    this.broadcastMessage(message, excluded)
+  }
+
+  broadcastMessage(message, excluded = []) {
+    Object.keys(this.peers).forEach(p => {
+      if (
+        this.peers[p] &&
+        this.peers[p].state === 'connected' &&
+        excluded.indexOf(p) === -1
+      ) {
+        this.peers[p].send('msg', message)
+      }
+    })
+  }
+
+  handleMessage(rawData, socket) {
+    const msg = JSON.parse(rawData.toString('utf8'))
+    let sender
+    let data
+
+    switch (msg.type) {
+      case 'REQ:BLOCKS':
+        sender = this.peers[msg.from]
+        if (sender) {
+          const fromBlock = +msg.fromBlock
+          const p = []
+          if (fromBlock > 0) {
+            for (let i = fromBlock; i < fromBlock + 5; i++) {
+              p.push(this.chain.getBlock(i))
+            }
+
+            Promise.all(p)
+              .then((result = []) => {
+                result = result
+                  .filter(r => r)
+                  .map(r => utils.bufferToHex(r.serialize()))
+
+                // send blocks
+                sender.send(
+                  'msg',
+                  JSON.stringify({
+                    type: 'RES:BLOCKS',
+                    from: this.hostString,
+                    data: result
+                  })
+                )
+              })
+              .catch(e => {
+                // supress error if any
+              })
+          }
+        }
+        break
+      case 'RES:BLOCKS':
+        data = msg.data || []
+        this.chain.putBlocks(
+          data.map(d => {
+            return new Block(utils.toBuffer(d))
+          })
+        )
+        break
+      case 'RES:PEERS':
+        data = msg.data || []
+        // add new peers
+        data.forEach(d => {
+          this.addPeer(d)
+        })
+        break
+      case 'REQ:PEERS':
+        sender = this.peer[msg.from]
+        if (sender) {
+          sender.send(
+            'msg',
+            JSON.stringify({
+              type: 'RES:PEERS',
+              from: this.hostString,
+              data: Object.keys(this.peers)
+            })
+          )
+        }
+        break
+      case 'TX:ADD':
```

Comment:
This key follows a different convention than the rest. It's noun:verb while others are {req,res}:noun.